### PR TITLE
Feat/redis enjoyment

### PR DESCRIPTION
**Pull request by kubapoke**
Added redis into our system. When offers are looked up, they're created as temporary objects in the redis database with uniquely generated Ids. Then, when an user proceeds to choose an offer, the back-end retrieves all of the informations about a particular offer from redis. Remember to update your CarRental .env file!

### CODE REVIEW

**Comment by kubapoke** [comment]
Before merging, I will need to fix pagination - it experiences a big lag which leads do other errors - most likely doe to Redis delay

**Comment by kubapoke** [comment]
I will reopen once the fix is ready

**Comment by kubapoke** [comment]
Everything works now, @Egor332 could you check it out?

File: car-rental/backend/CarRentalAPI/CarRentalAPI/Abstractions/Repositories/IOfferRepository.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using CarRentalAPI.DTOs.Redis;
+using CarRentalAPI.Models;
+
+namespace CarRentalAPI.Abstractions.Repositories
+{
+    public interface IOfferRepository
+    {
+        public Task<List<OfferForCarSearchDto>> CreateAndRetrieveOffersAsync(List<Car> cars, DateTime startDate, DateTime endDate,
+            string conditions, string companyName, string email);
+        public Task<OfferForRedisDto?> GetAndDeleteOfferAsync(string offerId);
```
Comment (Egor332) [inline car-rental/backend/CarRentalAPI/CarRentalAPI/Abstractions/Repositories/IOfferRepository.cs:11]: Get and delete? Isn't it violation of single responsibility?

**Comment by MegaRoboNinja** [comment]
Wow great job! It's impressive what you have done here. The team appreciates your sacrifice! We have not missed the fact that your last PR was submitted at 3:40 in the morning. 🫡

**Pull request merged**